source/AI.cpp

```diff
@@ -3292,20 +3292,40 @@ void AI::MovePlayer(Ship &ship, const PlayerInfo &player, Command &activeCommand
 			if(shift)
 				ship.SetTargetShip(shared_ptr<Ship>());
 
-			double closest = numeric_limits<double>::infinity();
+			double best = numeric_limits<double>::infinity();
 			bool foundEnemy = false;
 			bool foundAnything = false;
+			bool distancePriority = Preferences::Has("Board target");
+
+			auto strategy = [&](const AI &ai) -> function<double(const shared_ptr<Ship> &other)> {
```

Comment:
You don't need to do the `this` indirection.

Reply:
Are you sure?
I assume that by that you mean the (*this) at the end? If I don't include it, how do I call the function?

Reply:
`Has()`. The `this` is captured by the reference capture.

Reply:
I had to write strategy()(other) instead of strategy(other), blew my mind